src/Relations/BelongsToThrough.php

```diff
@@ -121,7 +138,24 @@ public function getForeignKeyName(Model $model = null)
             return $this->foreignKeyLookup[$table];
         }
 
-        return Str::singular($table).'_id';
+        return Str::singular($table) . '_id';
+    }
+
+    /**
+     * Get the local key for a model.
+     *
+     * @param \Illuminate\Database\Eloquent\Model|null $model
+     * @return string
+     */
+    public function getLocalKeyName(Model $model = null): string
+    {
+        $table = explode(' as ', ($model ?? $this->parent)->getTable())[0];
+
+        if (array_key_exists($table, $this->localKeyLookup)) {
+            return $this->localKeyLookup[$table];
+        }
+
+        return 'id';
```

Comment:
This is a breaking change: The fallback needs to be `getQualifiedKeyName()` as it was before (https://github.com/staudenmeir/belongs-to-through/blob/master/src/Relations/BelongsToThrough.php#L96)

Reply:
You are 100% correct, I have fixed it.